### PR TITLE
New Military Pilot Profession

### PR DESCRIPTION
**Pull request by Competently-Incompetent**
#38879 ## Summary
SUMMARY: content "Adds new profession Military Pilot"

#### Purpose of change
Most basic zombies have a profession counterpart and with the addition of flyable helicopters it felt like a good time to add another pilot profession.

#### Describe the solution
Added the military pilot profession. 
Costs 5 points.
Skills:
-Driving 4
-Mechanics 3
-Marksmanship 3
-Pistols 3
Starting equipment is based off the zombie military pilot loot drops.

#### Describe alternatives you've considered
Giving up and not adding this profession.

#### Testing
Selected the overrun scenario, selected the military pilot profession and spawned in. Did the same for the helicopter crash scenario. Did this several times for each scenario with no issues. 
Cdda Screenshot 2020 04 05 - 21 41 03 53
Cdda Screenshot 2020 04 05 - 18 15 41 56
Cdda Screenshot 2020 04 05 - 18 14 50 37

#### Additional context
To do:
-Add description (Done)
-Either edit the helicopter pilot trait to have a more generic description or add a new trait unique to the military pilot (Done, edited the existing trait description)
-add a leg ammo pouch with 2 magazines for the m17 (Had to do a workaround, it wouldn't let me spawn the leg ammo pouch with magazines so I've put them in the characters inventory instead)
-add to the DSA heli crash start (Done)

### CODE REVIEW

**Comment by carmakazi** [comment]
The M17 magazine ID is "p320mag_17rd_9x19mm"

File: src/handle_action.cpp [hunk 1]
```diff
@@ -96,6 +96,7 @@ static const bionic_id bio_remote( "bio_remote" );
 static const trait_id trait_HIBERNATE( "HIBERNATE" );
 static const trait_id trait_PROF_CHURL( "PROF_CHURL" );
 static const trait_id trait_PROF_HELI_PILOT( "PROF_HELI_PILOT" );
+static const trait_id trait_PROF_MILI_Pilot( "MILI_PILOT" );
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:99]:
```suggestion
static const trait_id trait_PROF_MILI_PILOT( "MILI_PILOT" );
```

File: src/handle_action.cpp [hunk 2]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
     if( !remote && z != 0 && !u.has_trait( trait_PROF_HELI_PILOT ) ) {
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:472]: Either you or I will get a merge conflict, we're both touching this line in separate PRs. Just a heads up. 
```suggestion
    if( !remote && z != 0 && ( !u.has_trait( trait_PROF_HELI_PILOT ) || !u.has_trait( trait_PROF_MILI_PILOT ) ) {
```

File: src/handle_action.cpp [hunk 3]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
...
         u.add_msg_if_player( m_info, _( "You have no idea how to make the vehicle fly." ) );
         return;
+    }
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:475]:
```suggestion
```

File: src/handle_action.cpp [hunk 4]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
...
+        }
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:476]:
```suggestion
```

File: src/handle_action.cpp [hunk 5]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
...
+    if( !remote && z != 0 && !u.has_trait( trait_MILI_PILOT ) ) {
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:477]:
```suggestion
```

File: src/handle_action.cpp [hunk 6]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
...
+        u.add_msg_if_player( m_info, _( "You have no idea how to make the vehicle fly." ) );
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:478]:
```suggestion
```

File: src/handle_action.cpp [hunk 7]
```diff
@@ -471,6 +472,11 @@ static void pldrive( int x, int y, int z = 0 )
...
+        return;
```
Comment (anothersimulacrum) [inline src/handle_action.cpp:479]:
```suggestion
```

File: src/handle_action.cpp (hunk 2 above)
Comment (Competently-Incompetent) [reply src/handle_action.cpp:472]: Ah, thanks for the warning. I'm new to all this, will these suggestions avoid the merge conflict?

Comment (anothersimulacrum) [reply src/handle_action.cpp:472]: Nope, I don't think we can avoid it.It should be quite easy to fix, but if you don't know what you're doing you can ask me and I can point you to advice.

Comment (Competently-Incompetent) [reply src/handle_action.cpp:472]: Any advice is appreciated, already feeling a little out of my depth

**Comment by LaVeyanFiend** [comment]
You should add this to the Dark Skies Ahead version of Heli-crash, as well.

(Pending permission from @ephemeralstoryteller)

File: data/json/professions.json [hunk 8]
```diff
@@ -4054,5 +4054,50 @@
       "male": [ "boxer_shorts", "pants", "dress_shoes", "polo_shirt", "socks" ],
       "female": [ "tank_top", "bra", "panties", "skirt", "heels", "jacket_leather", "stockings" ]
     }
-  }
+  },
+{
+  "type": "profession",
+  "ident": "mili_pilot",
+  "name": "Military Pilot",
+  "description": "You got to see things fall apart from the sky, transporting soldiers and survivors from one holdout to the next. You knew it was only a matter of time before the horrors patrolling the skies shot you down.",
```
Comment (Night-Pryanik) [inline data/json/professions.json:4062]:
```suggestion
  "description": "You got to see things fall apart from the sky, transporting soldiers and survivors from one holdout to the next.  You knew it was only a matter of time before the horrors patrolling the skies shot you down.",
```

**Comment by T-rexsuper1** [comment]
This is awesome, can't wait for it to be merged!

**Comment by Competently-Incompetent** [comment]
@davidpwbrown is it ok for me to edit the Helicopter pilot mutation to have a more generic description?

**Comment by Competently-Incompetent** [comment]
Hmm, not sure why the last two tests are failing. I've had a look at the results but I'm sure how to fix, anyone willing to look at the results?

[log.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/4465637/log.txt)

Can't find a log for the travis test, just got this message: 
(CLANG=clang++-3.8 MODS=--mods=RL_Classes TEST_STAGE=1 CXXFLAGS="-Wno-error=unused-command-line-argument -D__extern_always_inline='extern __always_inline'")

**Comment by anothersimulacrum** [comment]
They failed due to an unrelated error.

**Comment by Competently-Incompetent** [comment]
That's a relief, is there anything I need to/can do?

**Comment by anothersimulacrum** [comment]
You can track down what's causing it and fix it, but there's nothing you need to do and it's not super easy to track down.

**Comment by Competently-Incompetent** [comment]
In that case I'll leave it, thanks for help

File: data/json/mutations/mutations.json [hunk 9]
```diff
@@ -6097,7 +6097,7 @@
     "id": "PROF_HELI_PILOT",
     "name": "Helicopter Pilot",
     "points": 0,
-    "description": "You are a commercial helicopter pilot, there are no paying passengers anymore, yet still you dream of taking your bird for a spin.",
+    "description": "You are a trained pilot, You wonder if you will ever have the chance to fly again.",
```
Comment (Night-Pryanik) [inline data/json/mutations/mutations.json:6100]:
```suggestion
    "description": "You are a trained pilot, you wonder if you will ever have the chance to fly again.",
```

**Pull request merged**